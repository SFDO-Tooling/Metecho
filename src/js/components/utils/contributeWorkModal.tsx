import Button from '@salesforce/design-system-react/components/button';
import Card from '@salesforce/design-system-react/components/card';
import Modal from '@salesforce/design-system-react/components/modal';
import RadioGroup from '@salesforce/design-system-react/components/radio-group';
import Radio from '@salesforce/design-system-react/components/radio-group/radio';
import React, { ChangeEvent, useCallback, useState } from 'react';
import { Trans, useTranslation } from 'react-i18next';

import mapSvg from '@/img/map-lg.svg?raw';
import { Illustration } from '@/js/components/utils';
import { Epic } from '@/js/store/epics/reducer';
import { Org } from '@/js/store/orgs/reducer';
import { Task } from '@/js/store/tasks/reducer';

export type OrgData = Pick<Org, 'id' | 'org_config_name'>;

export type ContributeCallback = (
  orgData: OrgData,
  {
    useExistingTask,
    createEpicLessTask,
  }: {
    useExistingTask?: boolean;
    createEpicLessTask?: boolean;
  },
) => void;

interface Props {
  epic?: Epic;
  task?: Task;
  isOpen: boolean;
  hasPermissions: boolean;
  orgData: OrgData;
  hasDevOrg?: boolean;
  closeModal: () => void;
  doContribute: ContributeCallback;
}

const ContributeWorkModal = ({
  epic,
  task,
  isOpen,
  hasPermissions,
  orgData,
  hasDevOrg,
  closeModal,
  doContribute,
}: Props) => {
  const { t } = useTranslation();
  const [useExistingTask, setUseExistingTask] = useState(!hasDevOrg);
  const [createEpicLessTask, setCreateEpicLessTask] = useState(false);

  const doClose = useCallback(() => {
    setUseExistingTask(!hasDevOrg);
    setCreateEpicLessTask(false);
    closeModal();
  }, [closeModal, hasDevOrg]);

  const handleSubmit = useCallback(() => {
    doContribute(orgData, { useExistingTask, createEpicLessTask });
    doClose();
  }, [doContribute, orgData, useExistingTask, createEpicLessTask, doClose]);

  let contents = null;
  if (!hasPermissions) {
    contents = (
      <Trans i18nKey="cannotContributeWork">
        <p>
          <b>
            You do not have “push” access to the GitHub repository for this
            Project,
          </b>{' '}
          and cannot contribute changes in Metecho.
        </p>
        <p>
          Contact an admin for the repository on GitHub to grant additional
          permissions, then refresh this page to try again.
        </p>
      </Trans>
    );
  } else if (task) {
    const handleTaskSelectChange = (event: ChangeEvent<HTMLInputElement>) => {
      if (event.target.value === 'true') {
        setUseExistingTask(true);
      } else {
        setUseExistingTask(false);
      }
    };

    contents = (
      <>
        {hasDevOrg ? (
          <Trans i18nKey="contributeWorkFromTaskWithDevOrg">
            <p>
              <b>To contribute the work you’ve done in your Scratch Org,</b>{' '}
              you’ll start by creating a new Task.
            </p>
            <p>
              You cannot convert your Scratch Org into the Dev Org for this Task
              because there is already an existing Dev Org.
            </p>
          </Trans>
        ) : (
          <p>
            <Trans i18nKey="contributeWorkFromTask">
              <b>To contribute the work you’ve done in your Scratch Org,</b>{' '}
              you’ll start by making this Org the Dev Org for <em>this</em> Task
              or a <em>new</em> Task.
            </Trans>
          </p>
        )}
        <RadioGroup
          assistiveText={{
            label: t('Select Task Type'),
            required: t('Required'),
          }}
          className="slds-p-left_none"
          name="task-contribute-work"
          required
          onChange={handleTaskSelectChange}
          disabled={hasDevOrg}
        >
          <Radio
            id="org-current-task"
            labels={{
              label: t('Convert Scratch Org into Dev Org for this Task'),
            }}
            checked={useExistingTask}
            name="task-contribute-work"
            value="true"
          />
          <Radio
            id="org-new-task"
            labels={{
              label: t('Convert Scratch Org into Dev Org on a new Task'),
            }}
            checked={!useExistingTask}
            name="task-contribute-work"
            value="false"
          />
        </RadioGroup>
      </>
    );
  } else if (epic) {
    contents = (
      <Trans i18nKey="contributeWorkFromEpic">
        <p>
          <b>To contribute the work you’ve done in your Scratch Org,</b> you’ll
          start by creating a new Task.
        </p>
        <p>
          Your Scratch Org will become the Dev Org for the newly created Task.
        </p>
      </Trans>
    );
  } else {
    const handleEpicLessTaskChange = (event: ChangeEvent<HTMLInputElement>) => {
      if (event.target.value === 'true') {
        setCreateEpicLessTask(true);
      } else {
        setCreateEpicLessTask(false);
      }
    };

    contents = (
      <>
        <Trans i18nKey="contributeWorkFromProject">
          <p>
            <b>To contribute the work you’ve done in your Scratch Org,</b> start
            by creating a new Task (with or without a new Epic).
          </p>
          <p>
            Your Scratch Org will become the Dev Org for the newly created Task.
          </p>
        </Trans>
        <RadioGroup
          assistiveText={{
            label: t('Select Task Type'),
            required: t('Required'),
          }}
          className="slds-p-left_none"
          name="project-contribute-work"
          required
          onChange={handleEpicLessTaskChange}
        >
          <Radio
            id="org-epic-and-task"
            labels={{
              label: t('Create a new Epic and Task'),
            }}
            checked={!createEpicLessTask}
            name="project-contribute-work"
            value="false"
          />
          <Radio
            id="org-epic-less-task"
            labels={{
              label: t('Create a new Task with no Epic'),
            }}
            checked={createEpicLessTask}
            name="project-contribute-work"
            value="true"
          />
        </RadioGroup>
      </>
    );
  }

  return (
    <Modal
      isOpen={isOpen}
      size="small"
      heading={t('Contribute Work from Scratch Org')}
      footer={[
        <Button key="cancel" label={t('Cancel')} onClick={doClose} />,
        hasPermissions ? (
          <Button
            key="submit"
            label={t('Contribute')}
            variant="brand"
            onClick={handleSubmit}
          />
        ) : null,
      ]}
      prompt={hasPermissions ? undefined : 'warning'}
      dismissOnClickOutside={false}
      onRequestClose={doClose}
      assistiveText={{ closeButton: t('Cancel') }}
    >
      <div className="slds-grid slds-wrap slds-p-around_medium">
        <div
          className="slds-p-around_small
            slds-size_1-of-1
            slds-small-size_2-of-5"
        >
          <Card
            className="slds-card_boundary"
            bodyClassName="slds-card__body_inner"
            hasNoHeader
          >
            <Illustration svg={mapSvg} />
          </Card>
        </div>
        <div
          className="slds-p-around_small
            slds-size_1-of-1
            slds-small-size_3-of-5
            slds-text-longform"
        >
          {contents}
        </div>
      </div>
    </Modal>
  );
};

export default ContributeWorkModal;
