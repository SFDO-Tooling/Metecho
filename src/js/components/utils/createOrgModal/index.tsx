import Button from '@salesforce/design-system-react/components/button';
import Modal from '@salesforce/design-system-react/components/modal';
import React, { MouseEvent, useState } from 'react';
import { useTranslation } from 'react-i18next';

import {
  LabelWithSpinner,
  useForm,
  useFormDefaults,
  useIsMounted,
} from '@/js/components/utils';
import CreateOrgForm from '@/js/components/utils/createOrgModal/form';
import Overview from '@/js/components/utils/createOrgModal/overview';
import { Epic } from '@/js/store/epics/reducer';
import { Project } from '@/js/store/projects/reducer';
import { Task } from '@/js/store/tasks/reducer';
import {
  DEFAULT_ORG_CONFIG_NAME,
  OBJECT_TYPES,
  ORG_TYPES,
} from '@/js/utils/constants';

interface Props {
  project: Project;
  epic?: Epic;
  task?: Task;
  isOpen: boolean;
  closeModal: () => void;
}

export interface OrgData {
  description: string;
  org_config_name: string;
}

const CreateOrgModal = ({ project, epic, task, isOpen, closeModal }: Props) => {
  const { t } = useTranslation();
  const [pageIndex, setPageIndex] = useState(0);
  const isMounted = useIsMounted();
  const [isSaving, setIsSaving] = useState(false);

  const nextPage = () => {
    setPageIndex(pageIndex + 1);
  };

  const handleSuccess = () => {
    /* istanbul ignore else */
    if (isMounted.current) {
      setIsSaving(false);
      closeModal();
      setPageIndex(0);
    }
  };

  /* istanbul ignore next */
  const handleError = () => {
    if (isMounted.current) {
      setIsSaving(false);
    }
  };

  const defaultConfigName = task?.org_config_name || DEFAULT_ORG_CONFIG_NAME;
  const additionalData: { [key: string]: any } = {
    org_type: ORG_TYPES.PLAYGROUND,
  };

  if (task) {
    additionalData.task = task.id;
  } else if (epic) {
    additionalData.epic = epic.id;
  } else {
    additionalData.project = project.id;
  }

  const {
    inputs,
    errors,
    handleInputChange,
    setInputs,
    handleSubmit,
    resetForm,
  } = useForm({
    fields: {
      description: '',
      org_config_name: defaultConfigName,
    } as OrgData,
    additionalData,
    objectType: OBJECT_TYPES.ORG,
    onSuccess: handleSuccess,
    onError: handleError,
    shouldSubscribeToObject: false,
  });

  const handleClose = () => {
    closeModal();
    setPageIndex(0);
    resetForm();
  };

  // When parent org_config_name changes, update default selections
  useFormDefaults({
    field: 'org_config_name',
    value: defaultConfigName,
    inputs,
    setInputs,
  });

  const doSubmit = (e: MouseEvent<HTMLButtonElement>) => {
    setIsSaving(true);
    handleSubmit(e as any);
  };

  const CancelBtn = (
    <Button
      key="cancel"
      label={t('Cancel')}
      onClick={handleClose}
      disabled={isSaving}
    />
  );

  const pages = [
    {
      heading: t('Create Scratch Org'),
      contents: <Overview project={project} epic={epic} task={task} />,
      footer: [
        CancelBtn,
        <Button
          key="page-1-submit"
          label={t('Next')}
          variant="brand"
          onClick={nextPage}
        />,
      ],
    },
    {
      heading: t('Create Scratch Org'),
      contents: (
        <CreateOrgForm
          project={project}
          inputs={inputs as OrgData}
          errors={errors}
          handleInputChange={handleInputChange}
        />
      ),
      footer: [
        CancelBtn,
        <Button
          key="page-2-submit"
          label={
            isSaving ? (
              <LabelWithSpinner label={t('Creatingâ€¦')} variant="inverse" />
            ) : (
              t('Create Org')
            )
          }
          variant="brand"
          onClick={doSubmit}
          disabled={isSaving}
        />,
      ],
    },
  ];

  return (
    <Modal
      isOpen={isOpen}
      size="small"
      disableClose={isSaving}
      dismissOnClickOutside={false}
      heading={pages[pageIndex].heading}
      assistiveText={{ closeButton: t('Cancel') }}
      footer={pages[pageIndex].footer}
      onRequestClose={handleClose}
    >
      {pages[pageIndex].contents}
    </Modal>
  );
};

export default CreateOrgModal;
