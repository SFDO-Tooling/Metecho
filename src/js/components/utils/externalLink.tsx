import Icon from '@salesforce/design-system-react/components/icon';
import React, { ReactNode } from 'react';
import { useTranslation } from 'react-i18next';

import githubIcon from '@/img/github.svg';

const ExternalLink = ({
  url,
  showButtonIcon = false,
  showGitHubIcon = false,
  children,
  ...props
}: {
  url: string;
  showButtonIcon?: boolean;
  showGitHubIcon?: boolean;
  children?: ReactNode;
  [key: string]: any;
}) => {
  const { t } = useTranslation();
  return (
    <a href={url} target="_blank" rel="noreferrer noopener" {...props}>
      {showGitHubIcon && (
        <Icon
          path={`${githubIcon}#github`}
          size="xx-small"
          className="icon-link slds-m-bottom_xx-small"
        />
      )}
      {children}
      {showButtonIcon && (
        <Icon
          category="utility"
          name="new_window"
          size="xx-small"
          className="slds-button__icon slds-button__icon_right"
          containerClassName="slds-icon_container slds-current-color"
          assistiveText={{ label: t('Opens in new tab') }}
        />
      )}
    </a>
  );
};

export default ExternalLink;
