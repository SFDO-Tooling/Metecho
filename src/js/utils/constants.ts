export type ObjectTypes =
  | 'user'
  | 'project'
  | 'epic'
  | 'task'
  | 'scratch_org'
  | 'scratch_org_commit_metadata'
  | 'scratch_org_commit_dataset'
  | 'scratch_org_commit_omnistudio'
  | 'task_pr'
  | 'epic_pr'
  | 'issue'
  | 'dependency';

export const OBJECT_TYPES = {
  USER: 'user' as const,
  PROJECT: 'project' as const,
  EPIC: 'epic' as const,
  TASK: 'task' as const,
  ORG: 'scratch_org' as const,
  COMMIT_METADATA: 'scratch_org_commit_metadata' as const,
  COMMIT_DATASET: 'scratch_org_commit_dataset' as const,
  COMMIT_OMNISTUDIO: 'scratch_org_commit_omnistudio' as const,
  TASK_PR: 'task_pr' as const,
  EPIC_PR: 'epic_pr' as const,
  GITHUB_ISSUE: 'issue' as const,
  PROJECT_DEPENDENCY: 'dependency' as const,
};

export const DEFAULT_ORG_CONFIG_NAME = 'dev';

export const NULL_FILTER_VALUE = 'null';

export type OrgTypes = 'Dev' | 'QA' | 'Playground';

export const ORG_TYPES = {
  DEV: 'Dev' as const,
  QA: 'QA' as const,
  PLAYGROUND: 'Playground' as const,
};

export type WebsocketActions = 'SUBSCRIBE' | 'UNSUBSCRIBE';

export const WEBSOCKET_ACTIONS = {
  SUBSCRIBE: 'SUBSCRIBE' as const,
  UNSUBSCRIBE: 'UNSUBSCRIBE' as const,
};

export type TaskStatuses = 'Planned' | 'In progress' | 'Completed' | 'Canceled';

export const TASK_STATUSES = {
  PLANNED: 'Planned' as const,
  IN_PROGRESS: 'In progress' as const,
  COMPLETED: 'Completed' as const,
  CANCELED: 'Canceled' as const,
};

export type ReviewStatuses = 'Approved' | 'Changes requested';

export const REVIEW_STATUSES = {
  APPROVED: 'Approved' as const,
  CHANGES_REQUESTED: 'Changes requested' as const,
};

export type EpicStatuses = 'Planned' | 'In progress' | 'Review' | 'Merged';

export const EPIC_STATUSES = {
  PLANNED: 'Planned' as const,
  IN_PROGRESS: 'In progress' as const,
  REVIEW: 'Review' as const,
  MERGED: 'Merged' as const,
};

export const LIST_CHANNEL_ID = 'list';

export type OrgParentType = 'TASK' | 'EPIC' | 'PROJECT';

export type WalkthroughType = 'play' | 'help' | 'plan';
export const WALKTHROUGH_TYPES = {
  PLAY: 'play' as const,
  HELP: 'help' as const,
  PLAN: 'plan' as const,
};

export const SHOW_WALKTHROUGH = 'show_walkthrough';

export type ConfirmOrgTracker = 'delete' | 'refresh' | null;

export const CONFIRM_ORG_TRACKER = {
  DELETE: 'delete' as const,
  REFRESH: 'refresh' as const,
};

export const RETRIEVE_CHANGES = 'retrieve_changes';

export const CREATE_TASK_FROM_ORG = 'create_task_from_org';

export const TASKS_BY_PROJECT_KEY = 'all';

export const SHOW_PROJECT_CREATE_ERROR = 'show_project_create_error';
