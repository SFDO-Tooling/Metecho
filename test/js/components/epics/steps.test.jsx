import { pick } from 'lodash';
import React from 'react';

import EpicStatusSteps from '@/js/components/epics/steps';
import { EPIC_STATUSES, TASK_STATUSES } from '@/js/utils/constants';

import { render } from './../../utils';

const jonny = {
  id: 111111,
  login: 'jonny',
  permissions: { push: true },
};
const stacy = {
  id: 999999,
  login: 'stacy',
  permissions: { push: true },
};
const defaultEpic = {
  id: 'epic1',
  slug: 'epic-1',
  name: 'Epic 1',
  project: 'p1',
  branch_url: 'https://github.com/test/test-repo/tree/branch-name',
  branch_name: 'branch-name',
  github_users: [jonny.id, stacy.id],
  has_unmerged_commits: false,
  status: EPIC_STATUSES.PLANNED,
};
const epicWithTasks = {
  status: EPIC_STATUSES.IN_PROGRESS,
};
const epicWithChanges = {
  ...epicWithTasks,
  has_unmerged_commits: true,
};
const epicWithPR = {
  ...epicWithChanges,
  pr_is_open: true,
  pr_url: 'https://github.com/test/test-repo/pulls/123',
  status: EPIC_STATUSES.REVIEW,
};
const mergedEpic = {
  ...epicWithChanges,
  status: EPIC_STATUSES.MERGED,
};
const defaultTask = {
  id: 'task',
  epic: pick(defaultEpic, ['id', 'slug', 'name', 'github_users']),
  review_valid: false,
  review_status: '',
  pr_is_open: false,
  status: TASK_STATUSES.PLANNED,
  assigned_dev: null,
  assigned_qa: null,
  has_unmerged_commits: false,
  commits: [],
  origin_sha: 'parent_sha',
};
const taskWithDev = {
  assigned_dev: jonny,
  status: TASK_STATUSES.IN_PROGRESS,
};

describe('<EpicStatusSteps />', () => {
  test.each([
    ['planned, no tasks', {}, null],
    ['task created, no assigned dev', epicWithTasks, {}],
    ['dev assigned', epicWithTasks, taskWithDev],
    ['epic with commits', epicWithChanges, taskWithDev, true],
    ['epic with pr', epicWithPR, taskWithDev],
    ['merged epic', mergedEpic, taskWithDev],
  ])('renders steps: %s', (name, epicOpts, taskOpts, readyToSubmit = false) => {
    const epic = { ...defaultEpic, ...epicOpts };
    let tasks = [{ ...defaultTask, ...taskOpts }];
    if (taskOpts === null) {
      tasks = [];
    }
    const { container } = render(
      <EpicStatusSteps
        epic={epic}
        tasks={tasks}
        readyToSubmit={readyToSubmit}
        canSubmit
        handleAction={jest.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
