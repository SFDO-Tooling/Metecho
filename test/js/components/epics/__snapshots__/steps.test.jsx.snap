// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpicStatusSteps /> renders steps: dev assigned 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-active"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        >
          <span
            class="slds-assistive-text"
          >
            Active
          </span>
        </div>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Submit this Epic for review on GitHub
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Merge pull request on GitHub
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<EpicStatusSteps /> renders steps: epic with commits 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-active"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        >
          <span
            class="slds-assistive-text"
          >
            Active
          </span>
        </div>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          <button
            class="slds-button_reset slds-text-link"
            type="button"
          >
            Submit this Epic for review on GitHub
          </button>
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Merge pull request on GitHub
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<EpicStatusSteps /> renders steps: epic with pr 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Submit this Epic for review on GitHub
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-active"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        >
          <span
            class="slds-assistive-text"
          >
            Active
          </span>
        </div>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          <a
            href="https://github.com/test/test-repo/pulls/123"
            rel="noreferrer noopener"
            target="_blank"
          >
            Merge pull request on GitHub
            <span
              class="slds-m-left_xx-small"
            >
              <svg
                aria-hidden="true"
                class="slds-m-bottom_xx-small slds-icon slds-icon_xx-small slds-icon-text-default"
              >
                <use />
              </svg>
            </span>
          </a>
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<EpicStatusSteps /> renders steps: merged epic 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Submit this Epic for review on GitHub
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Merge pull request on GitHub
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<EpicStatusSteps /> renders steps: planned, no tasks 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-active"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        >
          <span
            class="slds-assistive-text"
          >
            Active
          </span>
        </div>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Submit this Epic for review on GitHub
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Merge pull request on GitHub
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<EpicStatusSteps /> renders steps: task created, no assigned dev 1`] = `
<div>
  <h3
    class="slds-text-heading_medium slds-m-bottom_small"
  >
    Next Steps for this Epic
  </h3>
  <div
    class="slds-progress slds-progress_vertical"
  >
    <ol
      class="slds-progress__list"
    >
      <li
        class="slds-progress__item slds-is-completed"
      >
        <span
          class="slds-icon_container slds-icon-utility-success slds-progress__marker slds-progress__marker_icon slds-m-left_small"
          title="Complete"
        >
          <svg
            aria-hidden="true"
            class="slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use />
          </svg>
          <span
            class="slds-assistive-text"
          >
            Complete
          </span>
        </span>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Create a Task
        </div>
      </li>
      <li
        class="slds-progress__item slds-is-active"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        >
          <span
            class="slds-assistive-text"
          >
            Active
          </span>
        </div>
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Assign a Developer to a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Complete a Task
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Submit this Epic for review on GitHub
        </div>
      </li>
      <li
        class="slds-progress__item"
      >
        <div
          class="slds-progress__marker slds-m-left_small"
        />
        <div
          class="slds-progress__item_content slds-grid slds-grid_align-spread"
        >
          Merge pull request on GitHub
        </div>
      </li>
    </ol>
  </div>
</div>
`;
